### DISCUSSION TITLE
Integrate Opentracing traces with opentelemetry traces

### DISCUSSION THREAD

**Discussion by pdkgamage** [open-telemetry/community#1043] [Community Discussions]
Hi All,

There are two micro services.
1. First micro service publishing traces using opentracing
2. Second micro service(implemented using vert.x) publishing traces using opentelemetry.

First micro service sending span-id and trace-id as header parameters. Is it possible to make parent span using those parameters In side the second micro service or any possible solution?

This is how registering opentelemetry in second micro service.

`SdkTracerProvider sdkTracerProvider = SdkTracerProvider.builder().build();
        OpenTelemetry openTelemetry = OpenTelemetrySdk.builder()
                .setTracerProvider(sdkTracerProvider)
                .setPropagators(ContextPropagators.create(W3CTraceContextPropagator.getInstance()))
                .buildAndRegisterGlobal();
                
        Vertx vertx = Vertx.vertx(new VertxOptions()
                .setTracingOptions(
                        new OpenTelemetryOptions(openTelemetry)
                ));`

Using following opentelemetry dependancy.

` <dependency>
            <groupId>io.vertx</groupId>
            <artifactId>vertx-opentelemetry</artifactId>
            <version>4.2.6</version>
        </dependency>`

Thanks,
Pradeep

**Comment by trask** [comment]
hi @pdkgamage! I'd recommend posting this to https://github.com/open-telemetry/opentelemetry-java
